src/Microsoft.TemplateEngine.Cli/CommandParsing/NewCommandInputCli.cs

```diff
@@ -0,0 +1,307 @@
+
+                // this must happen after ParseArgs(), which resets _templateParamCanonicalToVariantMap
+                _templateParamCanonicalToVariantMap = templateParamMap;
+
+                Dictionary<string, string> templateParamValues = new Dictionary<string, string>();
+
+                foreach (KeyValuePair<string, IReadOnlyList<string>> paramInfo in _templateParamCanonicalToVariantMap)
+                {
+                    string paramName = paramInfo.Key;
+                    string firstVariant = paramInfo.Value[0];
+
+                    // This returns true if the arg was specified, irrespective of whether it has a value.
+                    // If the arg was specified, it goes in the list. 
+                    // Null valued args are important - they facilitate bools & other value-optional args.
+                    if (_parseResult.TryGetArgumentValueAtPath(out string argValue, new[] { _commandName, firstVariant }))
+                    {
+                        templateParamValues.Add(paramName, argValue);
+                    }
+                }
+
+                _templateParamValues = templateParamValues;
+            }
+            catch (Exception ex)
+            {
+                throw new CommandParserException("Error parsing input parameters", string.Join(" ", _args), ex);
+            }
+        }
+
+        public string TemplateName => _templateNameArg;
+
+        public string Alias => _parseResult.GetArgumentValueAtPath(new[] { _commandName, "alias" });
+
+        public IList<string> ExtraArgsFileNames => _parseResult.GetArgumentListAtPath(new[] { _commandName, "extra-args" }).ToList();
+
+        public IList<string> ToInstallList => _parseResult.GetArgumentListAtPath(new[] { _commandName, "install" }).ToList();
+
+        public bool IsForceFlagSpecified => _parseResult.HasAppliedOption(new[] { _commandName, "force" });
+
+        public bool IsHelpFlagSpecified => _parseResult.HasAppliedOption(new[] { _commandName, "help" });
+
+        public bool IsListFlagSpecified => _parseResult.HasAppliedOption(new[] { _commandName, "list" });
+
+        public bool IsQuietFlagSpecified => _parseResult.HasAppliedOption(new[] { _commandName, "quiet" });
+
+        public bool IsShowAllFlagSpecified => _parseResult.HasAppliedOption(new[] { _commandName, "all" });
+
+        public string TypeFilter => _parseResult.GetArgumentValueAtPath(new[] { _commandName, "type" });
+
+        public string Language => _parseResult.GetArgumentValueAtPath(new[] { _commandName, "language" });
+
+        public string Locale => _parseResult.GetArgumentValueAtPath(new[] { _commandName, "locale" });
+
+        public string Name => _parseResult.GetArgumentValueAtPath(new[] { _commandName, "name" });
+
+        public string OutputPath => _parseResult.GetArgumentValueAtPath(new[] { _commandName, "output" });
+
+        public bool SkipUpdateCheck => _parseResult.HasAppliedOption(new[] { _commandName, "skip-update-check" });
```

Comment:
It's currently the only implementation now that I've removed the old, transitional version. But we decided to keep the interface so that if we decide to replace the parser, or try to use this with another front end, it's a pretty painless swap.